base/tps/src/org/dogtagpki/server/tps/rest/TokenService.java

```diff
@@ -310,11 +313,17 @@ protected void retrieveTokensWithVLV(
             Integer start,
             Integer size,
             TokenCollection response) throws Exception {
-
+//cfu here
         // search with VLV sorted by date in reverse order
         IDBVirtualList<TokenRecord> list = database.findRecords(
                 null, null, new String[] { "-modifyTimestamp", "-createTimestamp" }, size);
 
+        String userID = servletRequest.getUserPrincipal().getName();
+        CMS.debug("TokenService: principal name: " + userID);
+        IUGSubsystem userGroupManager = (IUGSubsystem) CMS.getSubsystem(CMS.SUBSYSTEM_UG);
+        IUser user = userGroupManager.getUser(userID);
+        List<String> authorizedProfiles = user.getTpsProfiles();
+
         int total = list.getSize();
```

Comment:
Since the total records returned is affected by `authorizedProfiles`, should the `total` be adjusted too?